### PR TITLE
view list of deals

### PR DESCRIPTION
**Pull request by amiaoh**
Deals tab shows a list of deals, including their
name
price per unit
date
store
Standard loading states

out of scope: create deals (create button does nothing)

### CODE REVIEW

File: app/types.tsx [hunk 1]
```diff
@@ -6,3 +6,14 @@ export type Store = {
 };
 
 export type Size = "large" | "small";
+
+export type DealAndStore = Deal & { store: Store };
```
Comment (yungyhan) [inline app/types.tsx]: `DealAndStore` kinda sounds like the type would be `Deal & Store`, probably more like `DealWithStore`

File: app/ViewDeals.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  //         onUpdateDealSuccess={async () => {
+  //           const result = await getDeals();
+  //           setDealList(result);
+  //           setDealBeingUpdated(undefined);
+  //         }}
+  //         deal={dealBeingUpdated}
+  //       ></UpdateDeal>
+  //     </Wrapper>
+  //   );
+  // }
```
Comment (yungyhan) [inline app/ViewDeals.tsx]: bad, ordinarily you should not leave this junk in but we'll let it slide this time

**Review by yungyhan — APPROVED** [review]
any code that gets merged to master should usually be stuff that you're happy for customers to see - here you had a `CreateDeal` button that does nothing, which surely would be a bug.

Make sure you only have code doing the thing that you're supposed to be doing next time

**Review by yungyhan — CHANGES_REQUESTED** [review]
actually let's change the name of the type

File: app/types.tsx (hunk 1 above)
Comment (amiaoh) [reply app/types.tsx]: https://github.com/auberginearms/groceryDatabaseApp/pull/26/commits/e0eccbfc1a04ae204b6f045e19ce39350d9ebe08

File: app/ViewDeals.tsx (hunk 2 above)
Comment (amiaoh) [reply app/ViewDeals.tsx]: https://github.com/auberginearms/groceryDatabaseApp/pull/26/commits/cacf23d8e39caeac6f435633507aa6b498f47cf6

Comment (amiaoh) [reply app/ViewDeals.tsx]: https://github.com/auberginearms/groceryDatabaseApp/pull/26/commits/53142294ecd6b2b814512951ecdbcb0b4bd30232

**Review by yungyhan — APPROVED** [review]

**Pull request merged**